apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-editor-preview.tsx

```diff
@@ -1,25 +1,36 @@
-import { useState } from 'react';
-import { RiArrowDownSLine, RiArrowUpSLine } from 'react-icons/ri';
+import { CSSProperties, useEffect, useRef, useState } from 'react';
+import { GeneratePreviewResponseDto } from '@novu/shared';
 
 import { Notification5Fill } from '@/components/icons';
 import { Code2 } from '@/components/icons/code-2';
 import { Button } from '@/components/primitives/button';
-import { Collapsible, CollapsibleContent, CollapsibleTrigger } from '@/components/primitives/collapsible';
 import { Editor } from '@/components/primitives/editor';
 import { InAppPreview } from '@/components/workflow-editor/in-app-preview';
-import { GeneratePreviewResponseDto } from '@novu/shared';
 import { loadLanguage } from '@uiw/codemirror-extensions-langs';
+import { Accordion, AccordionContent, AccordionItem, AccordionTrigger } from '@/components/primitives/accordion';
 
 type InAppEditorPreviewProps = {
   value: string;
   onChange: (value: string) => void;
   previewData?: GeneratePreviewResponseDto;
   applyPreview: () => void;
+  isPreviewLoading?: boolean;
 };
 export const InAppEditorPreview = (props: InAppEditorPreviewProps) => {
-  const { value, onChange, previewData, applyPreview } = props;
-  const [isEditorOpen, setIsEditorOpen] = useState(true);
+  const { value, onChange, previewData, applyPreview, isPreviewLoading } = props;
+  const [accordionValue, setAccordionValue] = useState('payload');
   const [payloadError, setPayloadError] = useState('');
+  const [height, setHeight] = useState(0);
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    setTimeout(() => {
+      if (contentRef.current) {
+        const rect = contentRef.current.getBoundingClientRect();
+        setHeight(rect.height);
+      }
+    }, 0);
+  }, [value]);
```

Comment:
have to calculate accordion height because the CodeMirror editor with `height: auto` works weird and updates the height after first render